### PR TITLE
OBPIH-2921 Fix calculation of QoH when inventory transactions involved

### PR DESCRIPTION
**Pull request by DominikaHatala**

### CODE REVIEW

**Comment by DominikaHatala** [comment]
@jmiranda I have found all occurrences of searching by inventory item instead of product and change it. Can you take a look?

**Comment by jmiranda** [comment]
Thanks @DominikaHatala. I am going to pull this down and review a little more closely tomorrow.

File: grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy [hunk 1]
```diff
@@ -879,13 +879,14 @@ class InventoryService implements ApplicationContextAware {
...
-        List transactionEntries = getTransactionEntriesByInventoryAndInventoryItems(location?.inventory, inventoryItems)
+        List products = inventoryItems.collect { it.product }
+        List transactionEntries = getTransactionEntriesByInventoryAndProduct(location?.inventory, products)
         List binLocations = getQuantityByBinLocation(transactionEntries)
         return binLocations
     }
 
     List getItemQuantityByBinLocation(Location location, InventoryItem inventoryItem) {
-        List transactionEntries = getTransactionEntriesByInventoryAndInventoryItem(location?.inventory, inventoryItem)
+        List transactionEntries = getTransactionEntriesByInventoryAndProduct(location?.inventory, [inventoryItem.product])
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy:889]: Now that we're return ALL transaction entries at the product level, we need to filter the resulting bin locations by inventoryItem.
```
binLocations = binLocations.findAll { it.inventoryItem == inventoryItem }
```

File: grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy [hunk 2]
```diff
@@ -879,13 +879,14 @@ class InventoryService implements ApplicationContextAware {
     }
 
     List getItemQuantityByBinLocation(Location location, List<InventoryItem> inventoryItems) {
-        List transactionEntries = getTransactionEntriesByInventoryAndInventoryItems(location?.inventory, inventoryItems)
+        List products = inventoryItems.collect { it.product }
+        List transactionEntries = getTransactionEntriesByInventoryAndProduct(location?.inventory, products)
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy:883]: Now that we're return ALL transaction entries at the product level, we need to filter the resulting bin locations by inventoryItem.
``` 
// pseudocode you might need to use a different mechanism to filter by inventory item 
binLocations = binLocations.findAll { it.inventoryItem in inventoryItems }
```

**Comment by jmiranda** [comment]
@awalkowiak can you take a look at this one too? If you think I went overboard removing dead code and refactoring methods, we can go back and focus on just the issue at hand which was related to a bug with ShipmentService.getQuantityOnHand() and ShipmentService.getTransactionEntries() methods ... not any of the ones in InventoryService. However, we should be trying to gut some of this unused code sooner or later.

**Comment by awalkowiak** [comment]
@jmiranda ok, I'll look into it today

**Comment by awalkowiak** [comment]
@jmiranda at first look it seems to be ok, but I think I'll still download the code and check it out locally.

**Comment by awalkowiak** [comment]
@jmiranda (but I won't have time to check it locally today)

**Comment by jmiranda** [comment]
@awalkowiak no worries. Run it locally and let me me know whether you think we should break it up into two separate PRs. I’m leaning that way myself. I can’t figure out a good way to test everything.

**Comment by awalkowiak** [comment]
@jmiranda I reviewed the code locally again (i was not testing it). I think it is ok. We can merge it but definitely without squashing.

**Pull request merged**